setup.py

```diff
@@ -5,38 +5,53 @@
     find_packages,
 )
 
-extras_require = {
-    'test': [
-        "pytest==3.3.2",
+
+def merge_lists(base_list, *other_lists):
+    new_list = base_list[:]
+
+    for other_list in other_lists:
+        new_list.extend(i for i in other_list if i not in new_list)
+
+    return new_list
+
+
+HYPOTHESIS_REQUIREMENT = "hypothesis>=3.6.1"
+
+extras_require = {}
+extras_require['tools'] = [
+    HYPOTHESIS_REQUIREMENT,
+]
+extras_require['test'] = merge_lists(
+    [
+        "pytest==4.4.1",
         "pytest-pythonpath>=0.7.1",
         "pytest-xdist==1.22.3",
-        "hypothesis>=3.6.1",
         "tox>=2.9.1,<3",
         "eth-hash[pycryptodome]",
+        HYPOTHESIS_REQUIREMENT,
     ],
-    'lint': [
-        "flake8==3.4.1",
-        "isort>=4.2.15,<5",
-        "mypy==0.620",
-    ],
-    'doc': [
-        "Sphinx>=1.6.5,<2",
-        "sphinx_rtd_theme>=0.1.9",
-    ],
-    'dev': [
+    extras_require['tools'],
+)
+extras_require['lint'] = [
+    "flake8==3.4.1",
+    "isort>=4.2.15,<5",
+    "mypy==0.620",
+]
+extras_require['doc'] = [
+    "Sphinx>=1.6.5,<2",
+    "sphinx_rtd_theme>=0.1.9",
+]
+extras_require['dev'] = merge_lists(
+    [
         "bumpversion>=0.5.3,<1",
         "pytest-watch>=4.1.0,<5",
         "wheel",
         "twine",
         "ipython",
     ],
-}
-
-extras_require['dev'] = (
-    extras_require['dev'] +
-    extras_require['test'] +
-    extras_require['lint'] +
-    extras_require['doc']
+    extras_require['test'],
```

Comment:
If you agree with the previous comment, maybe `tools` would be added here, in case things beyond hypothesis get added. NBD either way.